src/_P087_SerialProxy.ino

```diff
@@ -0,0 +1,402 @@
+#ifdef USES_P087
+
+// #######################################################################################################
+// #################### Plugin 087 Serial Proxy ##########################################################
+// #######################################################################################################
+//
+// Interact with a device connected to serial
+// Allows to redirect data to a controller
+//
+
+#include <ESPeasySerial.h>
+
+#define PLUGIN_087
+#define PLUGIN_ID_087          87
+#define PLUGIN_NAME_087       "Communication - Serial Proxy [TESTING]"
+
+
+#define P087_BAUDRATE       PCONFIG_LONG(0)
+#define P087_BAUDRATE_LABEL PCONFIG_LABEL(0)
+
+#define P087_QUERY_VALUE        0 // Temp placement holder until we know what selectors are needed.
+#define P087_NR_OUTPUT_OPTIONS  1
+
+#define P087_NR_OUTPUT_VALUES   1
+#define P087_QUERY1_CONFIG_POS  3
+
+#define P087_DEFAULT_BAUDRATE 38400
+
+#define P87_Nlines 2
+#define P87_Nchars 64
+
+#define P087_INITSTRING 0
+#define P087_EXITSTRING 1
+
+
+struct P087_data_struct : public PluginTaskData_base {
+  P087_data_struct() :  P087_easySerial(nullptr) {}
+
+  ~P087_data_struct() {
+    reset();
+  }
+
+  void reset() {
+    if (P087_easySerial != nullptr) {
+      delete P087_easySerial;
+      P087_easySerial = nullptr;
+    }
+  }
+
+  bool init(const int16_t serial_rx, const int16_t serial_tx, unsigned long baudrate) {
+    if ((serial_rx < 0) && (serial_tx < 0)) {
+      return false;
+    }
+    reset();
+    P087_easySerial = new ESPeasySerial(serial_rx, serial_tx);
+    P087_easySerial->begin(baudrate);
```

Comment:
@TD-er  should check for null here

Reply:
I will
Not sure if we could get a null pointer here though...